persistence/src/main/java/org/mskcc/smile/persistence/neo4j/TempoRepository.java

```diff
@@ -82,4 +83,52 @@ Tempo mergeMafCompleteEventBySamplePrimaryId(@Param("primaryId") String primaryI
             + "->(s:Sample)-[:HAS_METADATA]->(sm: SampleMetadata {primaryId: $primaryId}) "
             + "RETURN cc.date ORDER BY cc.date ASC LIMIT 1")
     String findInitialPipelineRunDateBySamplePrimaryId(@Param("primaryId") String primaryId);
+
+    @Query("MATCH (t:Tempo {accessLevel: $embargoAccessLevel}) WHERE date(t.embargoDate) < date() "
+            + "RETURN t.smileTempoId")
+    List<UUID> findTempoIdsNoLongerEmbargoed(@Param("embargoAccessLevel") String embargoAccessLevel);
+
+    @Query("MATCH (sm: SampleMetadata)<-[:HAS_METADATA]-(s: Sample)-[:HAS_TEMPO]->(t:Tempo) "
+            + "WHERE sm.primaryId IN $samplePrimaryIds SET t.accessLevel = $accessLevel")
+    void updateTempoAccessLevelBySamplePrimaryIds(@Param("samplePrimaryIds") List<String> samplePrimaryIds,
+            @Param("accessLevel") String accessLevel);
+
+    @Query("MATCH (s:Sample)-[:HAS_TEMPO]->(t:Tempo) "
+            + "WITH s, t, COLLECT { "
+            + "MATCH (s)-[:HAS_METADATA]->(sm:SampleMetadata) "
+            + "RETURN sm ORDER BY sm.importDate DESC LIMIT 1 "
+            + "} AS latestSm "
+            + "WITH s, t, latestSm[0] as latestSm "
```

Comment:
We might be able to shave off line 101 by modifying the statement above it as follows:
```neo4j
MATCH (s)-[:HAS_METADATA]->(sm:SampleMetadata) 
RETURN sm ORDER BY sm.importDate DESC LIMIT 1 
}[0] AS latestSm
```